src/Microsoft.Identity.Web/Policy/DenyGuestsAuthorizationRequirement.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Authorization;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Implements an <see cref="IAuthorizationRequirement"/>
+    /// which requires the current user to be a member of the tenant.
+    /// </summary>
+    public class DenyGuestsAuthorizationRequirement : IAuthorizationRequirement
```

Comment:
Not locked on the naming of this. I see many requirements are prefixed with `Require`, but `RequireTenantMembers` or `RequireNonGuests` sounds weird and isn't as explicit in what it tries to achieve.